Odata-docs/webapi-8/fundamentals/overview.md

```diff
@@ -0,0 +1,582 @@
+
+Here are examples of things that can be defined in an EDM model:
+- Entity types
+- Complex types
+- Enums
+- Entity Sets
+- Singletons
+- Functions
+- Actions
+
+The EDM model also supports type inheritance and relationships between types.
+
+Entity sets, singletons, functions and actions can be exposed as endpoints that you can make requests to. Entity types, complex types and enum types define types of objects that can be used as inputs or outputs of those endpoints.
+
+To learn more about the OData data model, [visit this article](/odata/concepts/data-model).
+
+By default, an OData service exposes the schema on the `GET /$metadata` endpoint. Here's an example of what a response from this endpoint looks like:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<edmx:Edmx Version="4.0" xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx">
+    <edmx:DataServices>
+        <Schema Namespace="Lab01.Models" xmlns="http://docs.oasis-open.org/odata/ns/edm">
+            <EntityType Name="Order">
+                <Key>
+                    <PropertyRef Name="Id" />
+                </Key>
+                <Property Name="Id" Type="Edm.Int32" Nullable="false" />
+                <Property Name="Amount" Type="Edm.Decimal" Nullable="false" Scale="Variable" />
+            </EntityType>
+            <EntityType Name="Customer">
+                <Key>
+                    <PropertyRef Name="Id" />
+                </Key>
+                <Property Name="Id" Type="Edm.Int32" Nullable="false" />
+                <Property Name="Name" Type="Edm.String" />
+                <NavigationProperty Name="Orders" Type="Collection(Lab01.Models.Order)" />
+            </EntityType>
+        </Schema>
+        <Schema Namespace="Default" xmlns="http://docs.oasis-open.org/odata/ns/edm">
+            <EntityContainer Name="Container">
+                <EntitySet Name="Customers" EntityType="Lab01.Models.Customer" />
+            </EntityContainer>
+        </Schema>
+    </edmx:DataServices>
+</edmx:Edmx>
+```
+
+Clients can consume this metadata to know what types and endpoints the service exposes. Many OData clients use this metadata to generate strongly-typed client code. Tools like the [OData Connected Service](/odata/connectedservice/getting-started) and [OData CLI](/odata/odatacli/getting-started) can generate client code based on an OData schema.
+
+## Registering OData services
+
+It is possible to service multiple OData APIs in the same ASP.NET Core application.
+
+The sample code in the introduction registers a single OData service using `options.AddRouteComponents(edmModel)`. This is hosted on the "default" route, i.e. on the root path `/`. This means that the endpoints on this OData service look like :
```

Comment:
How would someone know where the introduction is? Maybe you could make this a section link... `The sample code in the [introduction](#introduction) section`. Though I don't see a section with "Introduction" as a title in the document

Reply:
Yeah, I initially said "above". I could add an introduction title. But what if I just said "at the beginning of this document"?

Reply:
Or, "our first sample code in this document...?"